src/firefly/js/tables/TableUtil.js

```diff
@@ -936,6 +937,47 @@ export function getColsByType(tblColumns=[], type=COL_TYPE.ALL) {
                         && (type === COL_TYPE.ALL || matcher(col)));
 }
 
+export function getColByUtype(tableModel, utype) {
+    return getColumns(tableModel).filter( (col) => col.utype === utype);
+}
+
+export function getColByUCD(tableModel, ucd) {
+    return getColumns(tableModel).filter( (col) => col.UCD === ucd);
+}
+
+/**
+ * returns a 2-part array, [release date column, datarights column] if this table
+ * contains proprietary data
+ * @param tableModel
+ * @returns {string[]}
+ */
+export function getProprietaryInfo(tableModel) {
+    const rcname = get(tableModel, ['tableMeta', MetaConst.RELEASE_DATE_COL]);
+    const dcname = get(tableModel, ['tableMeta', MetaConst.DATARIGHTS_COL]);
+    return rcname || dcname ? [rcname, dcname] : [];
+}
+
+export function hasRowAccess(tableModel, rowIdx) {
```

Comment:
@deannaji This is Loi's nice little util function that can be called from your side to govern the preview tab layout.